dataRetriever/resolvers/peerAuthenticationResolver.go

```diff
@@ -0,0 +1,218 @@
+func (res *peerAuthenticationResolver) RequestDataFromReferenceAndChunk(hash []byte, chunkIndex uint32) error {
+	return res.SendOnRequestTopic(
+		&dataRetriever.RequestData{
+			Type:       dataRetriever.HashType,
+			Value:      hash,
+			ChunkIndex: chunkIndex,
+		},
+		[][]byte{hash},
+	)
+}
+
+// ProcessReceivedMessage will be the callback func from the p2p.Messenger and will be called each time a new message was received
+// (for the topic this validator was registered to, usually a request topic)
+func (res *peerAuthenticationResolver) ProcessReceivedMessage(message p2p.MessageP2P, fromConnectedPeer core.PeerID) error {
+	err := res.canProcessMessage(message, fromConnectedPeer)
+	if err != nil {
+		return err
+	}
+
+	res.throttler.StartProcessing()
+	defer res.throttler.EndProcessing()
+
+	rd, err := res.parseReceivedMessage(message, fromConnectedPeer)
+	if err != nil {
+		return err
+	}
+
+	switch rd.Type {
+	case dataRetriever.HashType:
+		return res.resolveOneHash(rd.Value, int(rd.ChunkIndex), message.Peer())
+	case dataRetriever.HashArrayType:
+		// Todo add implementation
+		err = dataRetriever.ErrRequestTypeNotImplemented
+	default:
+		err = dataRetriever.ErrRequestTypeNotImplemented
+	}
+	if err != nil {
+		err = fmt.Errorf("%w for value %s", err, logger.DisplayByteSlice(rd.Value))
+	}
+
+	return err
+}
+
+func (res *peerAuthenticationResolver) resolveOneHash(hash []byte, chunkIndex int, pid core.PeerID) error {
+	peerAuthMsgs := res.fetchPeerAuthenticationMessagesForHash(hash)
+	if len(peerAuthMsgs) == 0 {
+		return nil
+	}
+
+	if len(peerAuthMsgs) > maxNumOfPeerAuthenticationInResponse {
+		return res.sendMessageFromChunk(hash, peerAuthMsgs, chunkIndex, pid)
+	}
+
+	return res.marshalAndSend(&batch.Batch{Data: peerAuthMsgs}, pid)
+}
+
+func (res *peerAuthenticationResolver) sendMessageFromChunk(hash []byte, peerAuthMsgs [][]byte, chunkIndex int, pid core.PeerID) error {
+	maxChunks := len(peerAuthMsgs) / maxNumOfPeerAuthenticationInResponse
+	if len(peerAuthMsgs)%maxNumOfPeerAuthenticationInResponse != 0 {
+		maxChunks++
+	}
+
+	chunkIndexOutOfBounds := chunkIndex < 0 || chunkIndex > maxChunks
+	if chunkIndexOutOfBounds {
+		return nil
+	}
+
+	startingIndex := chunkIndex * maxNumOfPeerAuthenticationInResponse
+	endIndex := startingIndex + maxNumOfPeerAuthenticationInResponse
+	if endIndex > len(peerAuthMsgs) {
+		endIndex = len(peerAuthMsgs)
+	}
+	messagesBuff := peerAuthMsgs[startingIndex:endIndex]
+	chunk := batch.NewChunk(uint32(chunkIndex), hash, uint32(maxChunks), messagesBuff...)
+	return res.marshalAndSend(chunk, pid)
+}
+
+func (res *peerAuthenticationResolver) marshalAndSend(message *batch.Batch, pid core.PeerID) error {
+	buffToSend, err := res.marshalizer.Marshal(message)
+	if err != nil {
+		return err
+	}
+
+	return res.Send(buffToSend, pid)
+}
+
+func (res *peerAuthenticationResolver) fetchPeerAuthenticationMessagesForHash(hash []byte) [][]byte {
+	var messages [][]byte
+
+	keys := res.peerAuthenticationPool.Keys()
```

Comment:
considering the keys from the pool, might generate different views among nodes. 
Suggestion to do the following:
1. inject a nodes coordinator interface in this resolver
2. get all validators list. concatenate the list from each shard. sort validators based on public key
3. slice the resulting list based on the chunk index + maximum number of peers.
4. iterate the sub-slice and try to get (if existing) the peerAuth message from the pool